src/libsystemd/sd-journal/journal-file.c

```diff
@@ -326,7 +326,34 @@ static int journal_file_init_header(JournalFile *f, JournalFile *template) {
         return 0;
 }
 
+static int journal_file_refresh_boot_id(JournalFile *f, const sd_id128_t *boot_id) {
```

Comment:
(of course, adding such a logic would mean some special care needs to be taken when filtering by _BOOT_ID=/_MACHINE_ID=/_HOSTNAME=, since we have to skip whole files then, and never actually search for those fields.

Reply:
btw, along these lines, one more possible idea: it might work to reduce the size of the two usec timestamps in the entries from 64bit to 32bit. And then store a 64bit "base" value for each in the journal file header, so that the actual timestamps are always the sum of the base values plus the 32bit timestamps. And should the 32bit timestamps overrun, then use this as cause for rotation.

Reply:
(and similar for seqnum)

Reply:
> (Of course, this must be something that is indicated by a flag in the file, and that journald will enable but not the remoting stuff, as for that stuff multiple sources ight end up in the same journal file after all)

Adding another flag means a substantial amount of extra complexity, since we'll end up with 4 different versions of the Entry object struct (regular x multiple boot ids, regular x single boot id, compact x multiple boot ids, compact x single boot id).  I'd really like to keep things behind a single flag if possible. What if we put this behind compact mode as well? That would mean we can't use compact mode in combination with the journal remote stuff, but is that really a problem?